Buildings/Applications/DHC/EnergyTransferStations/CoolingDirectControlledReturn.mo

```diff
@@ -0,0 +1,379 @@
+    final m_flow_nominal={mDis_flow_nominal,-mBui_flow_nominal,mByp_flow_nominal},
+    dp_nominal=500*{1,-1,1})
+    "Bypass junction"
+    annotation (Placement(transformation(extent={{30,50},{50,70}})));
+
+  Buildings.Fluid.Sensors.MassFlowRate senMasFlo(
+    redeclare final package Medium = Medium)
+    "District supply mass flow rate sensor"
+    annotation (Placement(transformation(extent={{-50,50},{-30,70}})));
+
+  Buildings.Fluid.Sensors.TemperatureTwoPort senTDisSup(
+    redeclare final package Medium = Medium,
+    final m_flow_nominal=mDis_flow_nominal)
+    "District supply temperature sensor"
+    annotation (Placement(transformation(extent={{-90,50},{-70,70}})));
+
+  Buildings.Fluid.Sensors.TemperatureTwoPort senTDisRet(
+    redeclare final package Medium = Medium,
+    final m_flow_nominal=mDis_flow_nominal)
+    "District return temperature sensor"
+    annotation (Placement(transformation(extent={{-70,-70},{-50,-50}})));
+
+  Buildings.Fluid.Sensors.TemperatureTwoPort senTBuiRet(
+    redeclare final package Medium = Medium,
+    final m_flow_nominal=mBui_flow_nominal)
+    "Building return temperature sensor"
+    annotation (Placement(transformation(extent={{80,-70},{60,-50}})));
+
+  Modelica.Blocks.Continuous.Integrator int(final k=1)
+    "Integration"
+    annotation (Placement(transformation(extent={{70,100},{90,120}})));
+
+  Modelica.Blocks.Math.Add dTDis(final k1=-1,final k2=+1)
+    "Temperature difference on the district side"
+    annotation (Placement(transformation(extent={{-48,106},{-28,126}})));
+
+  Modelica.Blocks.Math.Product pro
+    "Product"
+    annotation (Placement(transformation(extent={{-10,100},{10,120}})));
+
+  Modelica.Blocks.Math.Gain cp(final k=cp_default)
+    "Specific heat multiplier to calculate heat flow rate"
+    annotation (Placement(transformation(extent={{30,100},{50,120}})));
+
+  Buildings.Fluid.FixedResistances.Junction spl(
+    redeclare final package Medium = Medium,
+    energyDynamics=Modelica.Fluid.Types.Dynamics.FixedInitial,
+    final m_flow_nominal={mBui_flow_nominal,-mDis_flow_nominal,-mByp_flow_nominal},
+    dp_nominal=500*{1,-1,1}) "Bypass junction, splitter"
+    annotation (Placement(transformation(extent={{50,-50},{30,-70}})));
+
+  Modelica.Fluid.Valves.ValveIncompressible cheVal(
+    redeclare final package Medium = Medium,
+    final allowFlowReversal=false,
+    final dp_nominal=dpCheVal_nominal,
+    final m_flow_nominal=mByp_flow_nominal,
+    final filteredOpening=true,
+    riseTime=60,
+    final checkValve=true)
+    "Check valve (backflow preventer)"
+    annotation (Placement(transformation(extent={{-10,-10},{10,10}},
+        rotation=90,
+        origin={40,0})));
+
+  Modelica.Blocks.Sources.Constant ope(final k=1)
+    "Check valve is always open in the positive flow direction"
+    annotation (Placement(transformation(extent={{0,-10},{20,10}})));
+
+  Buildings.Controls.Continuous.PIDHysteresis con(
+    eOn=0,
+    eOff=-0.5,
```

Comment:
should `eOn` and `eOff` be tunning parameters that user can change? If they are, then they should be propagated.

Reply:
Yes, I don't see why not. I'll propagate these as well.